File: examples/tutorials/array/usage/python/array_usage.py
from example_lib.array.array import Array

array = Array()

assert array.empty is True

array.push_back(0)
array.push_back(1)
array.push_back(2)

assert array.empty is False
assert array.size == 3
assert array.front() == 0
assert array.back() == 2
assert array[1] == 1

try:
    array.at(5)
    assert False
except IndexError:
    pass
except Exception:
    assert False

File: examples/tutorials/array/usage/swift/array_usage.swift
import XCTest
import Wrapper

func runArrayTests() {

    let array = Array()

    assert(array.empty)

    array.push_back(element: 0)
    array.push_back(element: 1)
    array.push_back(element: 2)

    assert(!array.empty)
    assert(array.size == 3)
    assert(array.front() == 0)
    assert(array.back() == 2)
    assert(array[1] == 1)

    do {
        let _ = try array.at(i: 5)
        assert(false)
    } catch is StdOutOfRange {
    } catch {
        assert(false)
    }

}

#if os(Linux)
runArrayTests()
#elseif os(OSX)
class ArrayTests: XCTestCase {
    func testRun() throws {
        runArrayTests()
    }
}
#endif
